experimental/dds/tree2/src/test/feature-libraries/modularChangeFamilyIntegration.spec.ts

```diff
@@ -99,6 +99,69 @@ describe("ModularChangeFamily integration", () => {
 			const expectedDelta = normalizeDelta(family.intoDelta(makeAnonChange(expected)));
 			assert.deepEqual(rebasedDelta, expectedDelta);
 		});
+
+		it("Nested moves both requiring a second pass", () => {
+			const [changeReceiver, getChanges] = testChangeReceiver(family);
+			const editor = new DefaultEditBuilder(family, changeReceiver);
+
+			const fieldAPath = { parent: undefined, field: fieldA };
+
+			// Note that these are the paths before any edits have happened.
+			const node1Path = { parent: undefined, parentField: fieldA, parentIndex: 1 };
+			const node2Path = { parent: node1Path, parentField: fieldB, parentIndex: 1 };
+
+			editor.enterTransaction();
+
+			// Moves node2, which is a child of node1 to an earlier position in its field
+			editor
+				.sequenceField({
+					parent: node1Path,
+					field: fieldB,
+				})
+				.move(1, 1, 0);
+
+			// Moves node1 to an earlier position in the field
+			editor.sequenceField(fieldAPath).move(1, 1, 0);
+
+			// Modifies node2 so that both fieldA and fieldB have changes that need to be transfered
+			// from a move source to a destination during invert.
+			editor
+				.sequenceField({
+					parent: node2Path,
+					field: fieldC,
+				})
+				.delete(0, 1);
```

Comment:
Consider moving the creation of the modification above the creation of the moves since it preceedes them in the scenario being tested.